webapp/app.py

```diff
@@ -100,9 +110,49 @@ def api():
     Show the static api page
     """
 
+    base_url = "https://raw.githubusercontent.com/maas/maas-openapi-yaml/main"
+    yaml_file = "openapi2.yaml"
+
+    try:
+        definition = request.urlopen(f"{base_url}/{yaml_file}")
+    except error.HTTPError or error.URLError:
+        # Fall back to local yaml file if hosted yaml is unreachable
+        definition = open(yaml_file)
+    loaded_definition = load(definition, Loader)
+    tagged_definition = {}
+
+    for endpoint in loaded_definition["paths"]:
+        for method in loaded_definition["paths"][endpoint]:
+            if method != "parameters":
+                try:
+                    if {
+                        endpoint: loaded_definition["paths"][endpoint]
+                    } not in tagged_definition[
+                        loaded_definition["paths"][endpoint][method]["tags"][0]
+                    ]:
+                        tagged_definition[
+                            loaded_definition["paths"][endpoint][method][
+                                "tags"
+                            ][0]
+                        ] = [
+                            *tagged_definition[
+                                loaded_definition["paths"][endpoint][method][
+                                    "tags"
+                                ][0]
+                            ],
+                            {endpoint: loaded_definition["paths"][endpoint]},
+                        ]
+                except KeyError:
```

Comment:
Is catching an error really the best way to handle this situation? If we know which keys might be missing in the above code, it would be clearer and less error prone to test for them explicitly (`if key in dict: ... else: ...`)

Reply:
I'll get back to you on this one, I can't quite remember my reasoning and the day is coming to an end :joy: